thylint/thylint.py

```diff
@@ -0,0 +1,317 @@
+    # title
+    "Interactive diagnostic command",
+    # message
+    "This command is usually used interactively only and should only be checked in for "
+    "demonstration purposes.",
+    # word(ish) regexps that should be flagged as warnings
+    regexp_of([
+        'ML_val',
+        'class_deps',
+        'code_deps',
+        'code_thms',
+        'find_consts',
+        'find_theorems',
+        'find_unused_assms',
+        'full_prf',
+        'help',
+        'locale_deps',
+        'prf',
+        'print_ML_antiquotations',
+        'print_abbrevs',
+        'print_antiquotations',
+        'print_attributes',
+        'print_bnfs',
+        'print_bundles',
+        'print_case_translations',
+        'print_cases',
+        'print_claset',
+        'print_classes',
+        'print_codeproc',
+        'print_codesetup',
+        'print_coercions',
+        'print_commands',
+        'print_context',
+        'print_definitions',
+        'print_defn_rules',
+        'print_facts',
+        'print_induct_rules',
+        'print_inductives',
+        'print_interps',
+        'print_locale',
+        'print_locales',
+        'print_methods',
+        'print_options',
+        'print_orders',
+        'print_quot_maps',
+        'print_quotconsts',
+        'print_quotients',
+        'print_quotientsQ3',
+        'print_quotmapsQ3',
+        'print_record',
+        'print_rules',
+        'print_simpset',
+        'print_state',
+        'print_statement',
+        'print_syntax',
+        'print_term_bindings',
+        'print_theorems',
+        'print_theory',
+        'print_trans_rules',
+        'smt_status',
+        'thm_deps',
+        'thm_oracles',
+        'thy_deps',
+        'unused_thms',
+        'value',
+        'values',
+        'welcome',
+        'term',
+        'prop',
+        'thm',
+        'typ',
+    ])
+)
+
+find_proof_words = (
+    # title
+    "Proof/counterexample finder",
+    # message
+    "This command is usually used interactively only to find a proof or counter example, "
+    "but only its result should be checked in.",
+    # word(ish) regexps that should be flagged as warnings
+    regexp_of([
+        'nitpick',
+        'nunchaku',
+        'quickcheck',
+        'sledgehammer',
+        'solve_direct',
+        'try',
+        'try0',
+    ])
+)
+
+sorry_words = (
+    # title
+    "Unfinished proof",
+    # message
+    "This command indicates an unfinished or aborted proof.",
+    # word(ish) regexps that should be flagged as warnings
+    regexp_of([
+        '\\\<proof>',
+        'sorry',
+        'oops'
+    ])
+)
+
+axiom_words = (
+    # title
+    "Axioms",
+    # message
+    "Locales or definitions should usually be preferred to axioms, because axioms may make the "
+    "logic inconsistent. Merely introducing a new constant (without new laws) is fine, though.",
+    # word(ish) regexps that should be flagged as warnings
+    regexp_of([
+        'axiomatization',
```

Comment:
I guess most pull requests don't touch the existing axiomatizations, so it will be rare for one to show up in a diff. If it ever becomes a problem though it could be worth adding syntax for telling the linter to ignore part of a file.

Reply:
That one is probably harder, although possible. We're always fine to just ignore the output, though, we don't have to make it a requirement to reach zero annotations.

Reply:
I'll leave this one for another iteration.